### ISSUE TITLE
Retinanet random_flip_horizontal implementation - question

### ISSUE THREAD

**Issue by pentanol2** [keras-team/keras-io#353]
In examples/vision/retinanet.py line 322:
```
boxes = tf.stack(
    [1 - boxes[:, 2], boxes[:, 1], 1 - boxes[:, 0], boxes[:, 3]], axis=-1
)

```
The boxes tensor has the shape **(num_boxes , xmin, ymin , xmax , ymax)**
when flipping the box horizontally from left to right the xmin and xmax normalized coordinates have to be sutracted from 1. 
I the case above does not **_boxes[:,2] (3rd index)_** and **_boxes[:,0](1st index)_** respectively refer to ymin and  num_boxes. I see this step as illogic. And even the order of operations seems to be ambiguous.

 Could you explain correct me if I am wrong ? 
Source:
https://github.com/keras-team/keras-io/blob/15a93b58140e9c5b759ff5139af70f9805467b14/examples/vision/retinanet.py#L322

**Comment by srihari-humbarwadi** [comment]
The boxes have (N, 4) as their shape. Where N is number of boxes. You can access the first element of all the boxes ie xmin, using `boxes[:, 0]`. (Check the axis along which the tensor is being indexed)

**Comment by pentanol2** [comment]
> The boxes have (N, 4) as their shape. Where N is number of boxes. You can access the first element of all the boxes ie xmin, using `boxes[:, 0]`. (Check the axis along which the tensor is being indexed)

@srihari-humbarwadi Yes, I noticed that now. It was a misconception. But why subtracting xmax from one before xmin ?

**Comment by pentanol2** [comment]
> > The boxes have (N, 4) as their shape. Where N is number of boxes. You can access the first element of all the boxes ie xmin, using `boxes[:, 0]`. (Check the axis along which the tensor is being indexed)
> 
> @srihari-humbarwadi Yes, I noticed that now. It was a misconception. But why subtracting xmax from one before xmin ?

@srihari-humbarwadi Ah yes after subtraction the max will be min and the min will be a max. Thank you for your reply :)

**Comment by sachinprasadhs** [comment]
Hi, 
If your issue is resolved, could you please spare some time to close the issue.
For the latest update on the tutorial `Retinanet`, please follow the link here https://keras.io/examples/vision/retinanet/
